packages/resizin/src/upload/upload.ts

```diff
@@ -1,45 +1,78 @@
+import { defaults } from 'lodash';
 import fetch from 'isomorphic-fetch';
-import NodeFormData from 'form-data';
 import Promise from 'promise-polyfill';
+import getUniqueId from 'uuid/v1';
 
-const uploadImage = (serverUrl: string, apiKey: string, imageId: string, file: string) => {
+const type = {
+    IMAGE: 'image',
+    FILE: 'file',
+};
+
+const mimeByType = {
+    [type.IMAGE]: 'image/png',
+    [type.FILE]: 'application/octet-stream',
+};
+
+export type FileType = 'file' | 'image';
+
+interface UploadOptions {
+    fileType?: FileType;
+    mime?: string;
+    autoId?: boolean;
+}
+
+const defaultOptions = { fileType: type.IMAGE, autoId: false };
```

Comment:
Is there any reason why not to set it `true` by default?

Reply:
That's a good question and I would like to hear an opinion of original authors of the library. But you're right, it would be convenient to has it `true` by default. I will do some research about using the library and eventually change it.